src/realm/object-store/c_api/object.cpp

```diff
@@ -157,6 +157,24 @@ RLM_API const void* _realm_object_get_native_ptr(realm_object_t* obj)
     return static_cast<const Object*>(obj);
 }
 
+RLM_API realm_object_t* realm_object_freeze(const realm_object_t* live_object, const realm_t* frozen_realm)
+{
+    return wrap_err([&]() {
+        const auto& realm = *frozen_realm;
+        auto frozen_object = live_object->freeze(realm);
+        return new realm_object_t{std::move(frozen_object)};
+    });
+}
+
+RLM_API realm_object_t* realm_object_thaw(const realm_object_t* frozen_object, const realm_t* live_realm)
+{
+    return wrap_err([&]() {
+        const auto& realm = *live_realm;
+        auto live_object = frozen_object->thaw(realm);
```

Comment:
What happens if the object has been deleted in the live Realm? Should this function (and the other "thaw" functions) return nullptr in that case?

Reply:
I changed the underlying thaw methods for objects and lists to return empty optionals when the referenced object/list has been deleted.